### PR TITLE
[App Service] Fix #19491: `az functionapp show`: Add kind validation for show operation

### LINKED ISSUE #19491: `az functionapp show` should not show webapps
## Describe the bug
 `az functionapp show` can be used to show webapps. There should be a type check to prevent this.

**Command Name**
`az functionapp show`

**Errors:**
```
Shows webapp
```

## To Reproduce:

- `az functionapp show -n {webapp_name}`

## Expected Behavior
Should give an error that functionapp could not be found.

## Environment Summary
```
Windows-10-10.0.19043-SP0
Python 3.8.10
Installer: PIP

azure-cli 2.27.2

```

### PR DESCRIPTION
**Pull request by parth21999**
**Description**
Fix #19491: `az functionapp show` should not show webapps.  

**Testing Guide**
`az functionapp show -n {webapp}` now returns an error message.
`az webapp show -n {functionapp}` now returns an error message.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Appservice

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -719,6 +715,33 @@ def list_function_app(cmd, resource_group_name=None):
...
+    external_kind = _kind_to_external_kind(app.kind) if app else None
+    if external_kind != kind:
+        raise ResourceNotFoundError(
+            "Unable to find {} '{}', in RG '{}'".format(kind, name, resource_group_name),
+            "Use 'az {} show' to show {}s".format(external_kind.lower(), external_kind))
+
+    app.site_config = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get_configuration', slot)
+    _rename_server_farm_props(app)
+    _fill_ftp_publishing_url(cmd, app, resource_group_name, name, slot)
+    print("ASFDALFH")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:733]: Why do we need to print this message?

**Comment by zhoxing-ms** [comment]
Because this is a bug fix, could you modify the PR title to start with `[]`? So that the PR title can be written into the release notes?

**Comment by zhoxing-ms** [comment]
Could you add some tests for this bug fix?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -719,6 +715,33 @@ def list_function_app(cmd, resource_group_name=None):
...
 
 
+def _show_app(cmd, resource_group_name, name, kind, slot=None):
+    app = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
+    if not app:
+        raise ResourceNotFoundError("Unable to find {} '{}', in RG '{}'.".format(
+                                    kind, name, resource_group_name))
+
+    external_kind = _kind_to_external_kind(app.kind) if app else None
+    if external_kind != kind:
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:725]: Can we make it so capitalization doesn't matter, or use an Enum?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:733]: oops, that was for debugging

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@parth21999 Could you please resolve this conflict?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**